### PR TITLE
kfp_tekton Integration Test

### PR DESCRIPTION
**Pull request by diegolovison**
kfp_tekton Integration Test

This PR includes the following changes:
* Refactor to use OpenShiftLibrary
* Refactor to support kfp_tekton
* kfp_tekton Integration Test

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -26,13 +32,8 @@ yq = "^3.1.0"
 pexpect = "^4.8.0"
 python-openstackclient = "^6.2.0"
 awscli = "^1.27.100"
-
-[tool.poetry.group.dev.dependencies]
-robotframework-lsp = "^1.8.0"
-robotframework-robocop = "^3.0.0"
-robotframework-tidy = "^3.3.3"
-black = ">20.0.0"
+kfp-tekton = "==1.5.3"
```
Comment (lugi0) [inline pyproject.toml:35]: This should go under dependencies

File: pyproject.toml [hunk 2]
```diff
@@ -8,6 +8,12 @@ packages = [{include = "ods_ci"}]
 
 [tool.poetry.dependencies]
 python = "^3.8"
+
+[tool.poetry.dev-dependencies]
+robotframework-lsp = "^1.8.0"
+robotframework-robocop = "^3.0.0"
+robotframework-tidy = "^3.3.3"
+black = ">20.0.0"
```
Comment (lugi0) [inline pyproject.toml:16]: these should be in `group.dev.dependencies`

File: pyproject.toml [hunk 3]
```diff
@@ -8,6 +8,12 @@ packages = [{include = "ods_ci"}]
 
 [tool.poetry.dependencies]
 python = "^3.8"
+
+[tool.poetry.dev-dependencies]
```
Comment (lugi0) [inline pyproject.toml:12]: this should be called `tool.poetry.dependencies`

File: pyproject.toml [hunk 4]
```diff
@@ -26,13 +32,8 @@ yq = "^3.1.0"
...
-[tool.poetry.group.dev.dependencies]
-robotframework-lsp = "^1.8.0"
-robotframework-robocop = "^3.0.0"
-robotframework-tidy = "^3.3.3"
-black = ">20.0.0"
+kfp-tekton = "==1.5.3"
 
 [build-system]
-requires = ["poetry-core"]
+requires = ["poetry"]
```
Comment (lugi0) [inline pyproject.toml:38]: why this change?

File: ods_ci/tests/Resources/Page/Pipelines/OpenShiftPipelines.robot [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+*** Keywords ***
+Data Science Pipelines Suite Setup
+    [Documentation]    Data Science Pipelines Suite Setup
+    Set Library Search Order    SeleniumLibrary
+    RHOSi Setup
+    Install Red Hat OpenShfit Pipelines
+
+Install Red Hat OpenShfit Pipelines
+    [Documentation]    Install Red Hat OpenShfit Pipelines
+    Oc Apply    kind=Subscription    src=${REDHAT_OPENSHIFT_PIPELINES_YAML}     namespace=openshift-operators
```
Comment (lugi0) [inline ods_ci/tests/Resources/Page/Pipelines/OpenShiftPipelines.robot:24]: if this YAML is the one used to create the pipelines operator in the cluster, I think the namespace should be `redhat-ods-operator`

File: ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/431__data-science-pipelines-api.robot [hunk 6]
```diff
@@ -49,22 +49,3 @@ Clear Data Science Pipelines Resources
     Delete Pipeline    ${pipeline_id}
     Remove Pipeline Project    ${project}
```
Comment (lugi0) [inline ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/431__data-science-pipelines-api.robot:50]: Isn't there an openshiftlibrary keyword to remove a namespace?

File: ods_ci/tests/Resources/Page/Pipelines/OpenShiftPipelines.robot [hunk 7]
```diff
@@ -0,0 +1,34 @@
+*** Settings ***
```
Comment (bdattoma) [inline ods_ci/tests/Resources/Page/Pipelines/OpenShiftPipelines.robot:1]: probably this file can go under `ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDataSciencePipelines.robot`

**Comment by lugi0** [comment]
@diegolovison I'm not sure what you mean by `poetry.lock is generated based on poetry lock`
After you update the dependencies in the toml file, you should either run `poetry install` (which will try to install the new dependencies without changing anything in `poetry.lock`) or `poetry update` (which will resolve the new dependencies and fetch all available updates for the libraries already in `poetry.lock` - equivalent to removing `poetry.lock` and running `poetry install`)

File: ods_ci/tests/Resources/Page/Pipelines/OpenShiftPipelines.robot [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+Data Science Pipelines Suite Setup
+    [Documentation]    Data Science Pipelines Suite Setup
+    Set Library Search Order    SeleniumLibrary
+    RHOSi Setup
+    Install Red Hat OpenShfit Pipelines
+
+Install Red Hat OpenShfit Pipelines
+    [Documentation]    Install Red Hat OpenShfit Pipelines
+    Oc Apply    kind=Subscription    src=${REDHAT_OPENSHIFT_PIPELINES_YAML}     namespace=openshift-operators
+    ${pod_count}    Get Redhat Openshift Pipelines
```
Comment (bdattoma) [inline ods_ci/tests/Resources/Page/Pipelines/OpenShiftPipelines.robot:25]: maybe a more clear name could be Get OpenShift Pipelines Pods Number

File: ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/431__data-science-pipelines-api.robot [hunk 9]
```diff
@@ -1,17 +1,17 @@
...
+Resource            ../../../Resources/Page/Pipelines/OpenShiftPipelines.robot
 Library             DateTime
 Library             ../../../../libs/DataSciencePipelinesAPI.py
 Suite Setup         Data Science Pipelines Suite Setup
 Suite Teardown      RHOSi Teardown
 
 
 *** Variables ***
-${REDHAT_OPENSHIFT_PIPELINES_YAML}=            ods_ci/tests/Resources/Files/redhat-openshift-pipelines.yaml
 ${DATA_SCIENCE_PIPELINES_APPLICATION_YAML}=    ods_ci/tests/Resources/Files/data-science-pipelines-sample.yaml
```
Comment (bdattoma) [inline ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/431__data-science-pipelines-api.robot:15]: should this be moved to the Resource file like done with line 14?

Comment (bdattoma) [reply ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/431__data-science-pipelines-api.robot:15]: it would also avoid to define this variable twice (here and in ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/432__data-science-pipelines-tekton.robot)

File: ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/432__data-science-pipelines-tekton.robot [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+Suite Teardown      RHOSi Teardown
+
+
+*** Variables ***
+${DATA_SCIENCE_PIPELINES_APPLICATION_YAML}=    ods_ci/tests/Resources/Files/data-science-pipelines-sample.yaml
+${URL_TEST_PIPELINE_RUN_YAML}=                 https://raw.githubusercontent.com/opendatahub-io/data-science-pipelines-operator/main/tests/resources/dsp-operator/test-pipeline-run.yaml
+
+
+*** Test Cases ***
+Verify ODS users can create and run a data science pipeline using the kfp_tekton python package
```
Comment (bdattoma) [inline ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/432__data-science-pipelines-tekton.robot:21]: could you remove "OSD" and make the title Camel Case?

File: ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/432__data-science-pipelines-tekton.robot [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+Resource            ../../../Resources/RHOSi.resource
+Resource            ../../../Resources/ODS.robot
+Resource            ../../../Resources/Common.robot
+Resource            ../../../Resources/Page/ODH/ODHDashboard/ODHDashboard.robot
+Resource            ../../../Resources/Page/Pipelines/OpenShiftPipelines.robot
+Library             DateTime
+Library             ../../../../libs/DataSciencePipelinesAPI.py
+Library             ../../../../libs/DataSciencePipelinesKfpTekton.py
+Suite Setup         Data Science Pipelines Suite Setup
+Suite Teardown      RHOSi Teardown
```
Comment (bdattoma) [inline ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/432__data-science-pipelines-tekton.robot:12]: I think we would need a teardown to clean up the cluster from the pipeline resources which were created by the tests

File: ods_ci/libs/DataSciencePipelinesKfpTekton.py [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+        self.api = None
+
+    def get_client(self, user, pwd, project):
+        if self.client is None:
+            self.api = DataSciencePipelinesAPI()
+            self.api.login_using_user_and_password(user, pwd, project)
+            self.client = kfp_tekton.TektonClient(
+                host=f'https://{self.api.route}/',
+                existing_token=self.api.sa_token,
+                ssl_ca_cert='/home/dlovison/github/diego-kitchen/rhods-qe-kfp-tekton/openshift.crt'
```
Comment (tarukumar) [inline ods_ci/libs/DataSciencePipelinesKfpTekton.py:84]: This should change right? you can not have hard coded path for certificate.

Also, is it tls certificate or some other certificate and how you captured the pod certificate. I am asking this because we can automate the whole process

File: ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/431__data-science-pipelines-api.robot (hunk 6 above)
Comment (bdattoma) [reply ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/431__data-science-pipelines-api.robot:50]: you could use `Oc Delete    kind=Project   name=${project_k8s_name}`

Comment (tarukumar) [reply ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/431__data-science-pipelines-api.robot:50]: We have keyword ` Oc Delete  kind=Project  name=<name of the project>`

File: ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/432__data-science-pipelines-tekton.robot [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+    ...         ????
+    End To End Pipeline Workflow Using Kfp_tekton    ${TEST_USER.USERNAME}    ${TEST_USER.PASSWORD}    pipelineskfptekton1
+
+
+*** Keywords ***
+End To End Pipeline Workflow Using Kfp_tekton
+    [Documentation]    Create, run and double check the pipeline result using Kfp_tekton python package. In the end,
+    ...    clean the pipeline resources.
+    [Arguments]     ${username}    ${password}    ${project}
+    Remove Pipeline Project    ${project}
```
Comment (tarukumar) [inline ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/432__data-science-pipelines-tekton.robot:35]: Same as earlier use openshift library

File: ods_ci/tests/Resources/Page/Pipelines/OpenShiftPipelines.robot (hunk 5 above)
Comment (tarukumar) [inline ods_ci/tests/Resources/Page/Pipelines/OpenShiftPipelines.robot:24]: You don't need namespace parameter since in your yaml file you have already declared the namespace under metadata section

https://github.com/red-hat-data-services/ods-ci/blob/master/ods_ci/tests/Resources/Files/redhat-openshift-pipelines.yaml

File: ods_ci/libs/DataSciencePipelinesAPI.py [hunk 14]
```diff
@@ -15,14 +15,6 @@ def __init__(self):
...
-    @keyword
-    def create_operator(self, file):
-        print(os.path.abspath(__file__))
-        current_path = __file__.index('/libs/')
-        current_path = __file__[:current_path]
-        response, error = self.run_oc(f"oc apply -f {current_path}/{file}")
-        print(f'create_operator({error}): {response}')
-
     @keyword
     def get_redhat_openshift_pipelines(self):
```
Comment (jgarciao) [inline ods_ci/libs/DataSciencePipelinesAPI.py:19]: In my opinion this method should be called `Wait Until OpenShift Pipelines Operator Is Deployed` and be added as keyword in ods_ci/tests/Resources/Operators/OpenShiftPipelines  (new file to have keywords related to this operator, see comment below)
	  
Alternativelly, we already have some keywords handling installation of operators in ods_ci/tests/Resources/Page/OCPDashboard/OperatorHub/OperatorHub.robot and ods_ci/tests/Resources/Page/OCPDashboard/InstalledOperators/InstalledOperators.robot

File: ods_ci/tests/Resources/Page/Pipelines/OpenShiftPipelines.robot (hunk 7 above)
Comment (jgarciao) [reply ods_ci/tests/Resources/Page/Pipelines/OpenShiftPipelines.robot:1]: This is related to the OpenShift Pipelines Operator, not Data Science Pipelines in RHODS. 

In my opinion, it could be in a new resource file to consolidate keywords to manage that operator: ods_ci/tests/Resources/Operators/OpenShiftPipelines

```
Install Red Hat OpenShift Pipelines Operator  version=1.9.0
Wait Until OpenShift Pipelines Operator Is Deployed
Is OpenShift Operator Installed
Get OpenShift Operator Version
```
Beware there are already methods to implement this using the UI in in ods_ci/tests/Resources/Page/OCPDashboard/OperatorHub/OperatorHub.robot and ods_ci/tests/Resources/Page/OCPDashboard/InstalledOperators/InstalledOperators.robot


For keywords related to RHODS Data Science Pipelines, I agree with @bdattoma that we could concentrate them in ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDataSciencePipelines.robot

File: ods_ci/tests/Resources/Page/Pipelines/OpenShiftPipelines.robot [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+Library             DateTime
+Library             ../../../../libs/DataSciencePipelinesAPI.py
+
+
+*** Variables ***
+${REDHAT_OPENSHIFT_PIPELINES_YAML}=            ods_ci/tests/Resources/Files/redhat-openshift-pipelines.yaml
+
+
+*** Keywords ***
+Data Science Pipelines Suite Setup
```
Comment (jgarciao) [inline ods_ci/tests/Resources/Page/Pipelines/OpenShiftPipelines.robot:16]: To me Suite Setup should be in the Test Suite, not here

File: ods_ci/tests/Resources/Page/Pipelines/OpenShiftPipelines.robot (hunk 5 above)
Comment (jgarciao) [reply ods_ci/tests/Resources/Page/Pipelines/OpenShiftPipelines.robot:24]: openshift-operators is correct here. This is installing the OpenShift Pipelines operator in the cluster (requirement for DAta Science Pipelines in RHODS). 

The Data Science Pipelines Operator is deployed automatically in RHODS 1.26 in redhat-ods-applications

File: ods_ci/tests/Resources/Page/Pipelines/OpenShiftPipelines.robot (hunk 8 above)
Comment (jgarciao) [reply ods_ci/tests/Resources/Page/Pipelines/OpenShiftPipelines.robot:25]: As suggested above, I think `Wait Until OpenShift Pipelines Operator Is Deployed` is clearer

File: ods_ci/tests/Resources/Page/Pipelines/OpenShiftPipelines.robot [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+    RHOSi Setup
+    Install Red Hat OpenShfit Pipelines
+
+Install Red Hat OpenShfit Pipelines
+    [Documentation]    Install Red Hat OpenShfit Pipelines
+    Oc Apply    kind=Subscription    src=${REDHAT_OPENSHIFT_PIPELINES_YAML}     namespace=openshift-operators
+    ${pod_count}    Get Redhat Openshift Pipelines
+    Should Be True    ${pod_count} == 1    All the pods were created
+
+Install DataSciencePipelinesApplication CR
```
Comment (jgarciao) [inline ods_ci/tests/Resources/Page/Pipelines/OpenShiftPipelines.robot:28]: As written above in another comment, we can consolidate keywords related to Data Science Pipelines in RHODS in ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDataSciencePipelines.robot

If the intent of this keyword is to create a DataSciencePipelineApplication, I think it could be renamed to `Create DataSciencePipelinesApplication` and have the yaml as argument

```
Create DataSciencePipelinesApplication
[Arguments]     ${project}  ${yaml}
```

File: ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/431__data-science-pipelines-api.robot [hunk 17]
```diff
@@ -49,22 +49,3 @@ Clear Data Science Pipelines Resources
     Delete Pipeline    ${pipeline_id}
     Remove Pipeline Project    ${project}
 
-Data Science Pipelines Suite Setup
```
Comment (jgarciao) [inline ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/431__data-science-pipelines-api.robot:52]: Suite setup was fine here

File: ods_ci/libs/DataSciencePipelinesKfpTekton.py (hunk 12 above)
Comment (diegolovison) [reply ods_ci/libs/DataSciencePipelinesKfpTekton.py:84]: Yeap. It will change

File: ods_ci/tests/Resources/Page/Pipelines/OpenShiftPipelines.robot (hunk 8 above)
[the remark this answers is not in the document]
Comment (diegolovison) [reply ods_ci/tests/Resources/Page/Pipelines/OpenShiftPipelines.robot:25]: No because there is a rule: https://github.com/red-hat-data-services/ods-ci/blob/master/ods_ci/libs/DataSciencePipelinesAPI.py#L27-L45

File: ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/432__data-science-pipelines-tekton.robot (hunk 13 above)
Comment (diegolovison) [reply ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/432__data-science-pipelines-tekton.robot:35]: Remove Pipeline Project will be addressed in other PR. The current code [double check ](https://github.com/red-hat-data-services/ods-ci/blob/master/ods_ci/libs/DataSciencePipelinesAPI.py#L84-L97) wait because it could be in Terminating status

File: ods_ci/libs/DataSciencePipelinesKfpTekton.py
Comment (jgarciao) [inline ods_ci/libs/DataSciencePipelinesKfpTekton.py:1]: This file is used to test just flip-coin pipeline. I think it's a good start, but a more versatile approach could be to create a keyword in ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDataSciencePipelines.robot called 

```
Run Pipeline Using KFP Tekton
[Arguments]    api_endpoint   bearer_token  pipeline_py
```
With this you could run most of [Trevor's examples](https://github.com/rh-intelligent-application-practice/kubeflow-examples/blob/main/pipelines/5_metrics_pipeline.py) just setting the environment variables and running the .py files


https://github.com/rh-intelligent-application-practice/kubeflow-examples/blob/main/pipelines/5_metrics_pipeline.py

**Review by jgarciao — CHANGES_REQUESTED** [review]

File: ods_ci/tests/Resources/Page/Pipelines/OpenShiftPipelines.robot (hunk 5 above)
Comment (lugi0) [reply ods_ci/tests/Resources/Page/Pipelines/OpenShiftPipelines.robot:24]: When I installed it for Elyra it got deployed in redhat-ods-operator

File: ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/432__data-science-pipelines-tekton.robot (hunk 10 above)
Comment (diegolovison) [reply ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/432__data-science-pipelines-tekton.robot]: Done

File: ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/431__data-science-pipelines-api.robot (hunk 6 above)
Comment (diegolovison) [reply ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/431__data-science-pipelines-api.robot:50]: Remove Pipeline Project will be addressed in other PR. The current code [double check ](https://github.com/red-hat-data-services/ods-ci/blob/master/ods_ci/libs/DataSciencePipelinesAPI.py#L84-L97)wait because it could be in Terminating status

File: ods_ci/tests/Resources/Page/Pipelines/OpenShiftPipelines.robot (hunk 5 above)
Comment (diegolovison) [reply ods_ci/tests/Resources/Page/Pipelines/OpenShiftPipelines.robot]: The right namespace is `openshift-operators`. It was removed from `Oc Apply` because it is inside the file (`REDHAT_OPENSHIFT_PIPELINES_YAML`)

File: ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/431__data-science-pipelines-api.robot (hunk 9 above)
Comment (diegolovison) [reply ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/431__data-science-pipelines-api.robot:15]: Done

File: ods_ci/tests/Resources/Page/Pipelines/OpenShiftPipelines.robot (hunk 7 above)
Comment (diegolovison) [reply ods_ci/tests/Resources/Page/Pipelines/OpenShiftPipelines.robot]: The files split is done

File: ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/432__data-science-pipelines-tekton.robot (hunk 11 above)
Comment (diegolovison) [reply ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/432__data-science-pipelines-tekton.robot:15]: Done

File: ods_ci/libs/DataSciencePipelinesKfpTekton.py (hunk 12 above)
Comment (diegolovison) [reply ods_ci/libs/DataSciencePipelinesKfpTekton.py]: Done

**Comment by diegolovison** [comment]
https://opendatascience-jenkins-csb-rhods.apps.ocp-c1.prod.psi.redhat.com/job/rhods-ci-pr-test/1720/

File: ods_ci/tests/Resources/Page/Operators/OpenShiftPipelines.resource [hunk 18]
```diff
@@ -0,0 +1,20 @@
...
+*** Variables ***
+${REDHAT_OPENSHIFT_PIPELINES_YAML}=            ods_ci/tests/Resources/Files/redhat-openshift-pipelines.yaml
+
+
+*** Keywords ***
+Install Red Hat OpenShift Pipelines
+    [Documentation]    Install Red Hat OpenShift Pipelines
+    Oc Apply    kind=Subscription    src=${REDHAT_OPENSHIFT_PIPELINES_YAML}
+    ${pod_count}    Wait Until OpenShift Pipelines Operator Is Deployed
+    Should Be True    ${pod_count} == 1    All the pods were created
```
Comment (bdattoma) [inline ods_ci/tests/Resources/Page/Operators/OpenShiftPipelines.resource:20]: is this file still required? I thought we replaced this in the PR #767 by @tarukumar

File: ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/432__data-science-pipelines-tekton.robot [hunk 19]
```diff
@@ -0,0 +1,47 @@
...
+    [Arguments]    ${username}    ${password}    ${project}
+    Remove Pipeline Project    ${project}
+    New Project    ${project}
+    Install DataSciencePipelinesApplication CR    ${project}
+    ${status}    Login Using User And Password    ${username}    ${password}    ${project}
+    Should Be True    ${status} == 200    DSP routing is working
+    ${result}    Kfp Tekton Create Run From Pipeline Func    ${username}    ${password}    ${project}    flipcoin_pipeline    # robocop: disable:line-too-long
+    ${run_status}   Kfp Tekton Wait For Run Completion    ${username}    ${password}    ${project}    ${result}
+    Should Be True    '${run_status}' == 'Completed'    Run ends
+    [Teardown]    Remove Pipeline Project    ${project}
```
Comment (bdattoma) [inline ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/432__data-science-pipelines-tekton.robot:41]: Teardown must be in the Test Case, not in the keyword

File: ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/431__data-science-pipelines-api.robot [hunk 20]
```diff
@@ -1,75 +1,51 @@
...
     Should Be True    '${run_status}' == 'Completed'    Run ends
-    [Teardown]    Clear Data Science Pipelines Resources    ${run_id}    ${pipeline_id}    ${project}
-
-Clear Data Science Pipelines Resources
-    [Documentation]    Cleans the pipeline resources.
-    [Arguments]     ${run_id}    ${pipeline_id}    ${project}
     Delete Runs    ${run_id}
     Delete Pipeline    ${pipeline_id}
-    Remove Pipeline Project    ${project}
+    [Teardown]    Remove Pipeline Project    ${project}
```
Comment (bdattoma) [inline ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/431__data-science-pipelines-api.robot:45]: Teardown must be in the Test Case, not in the keyword

**Comment by jgarciao** [comment]
Tested successfully here: https://opendatascience-jenkins-csb-rhods.apps.ocp-c1.prod.psi.redhat.com/job/rhods-ci-pr-test/1741/TestReport/

**Review by jgarciao — CHANGES_REQUESTED** [review]

**Comment by diegolovison** [comment]
https://opendatascience-jenkins-csb-rhods.apps.ocp-c1.prod.psi.redhat.com/job/rhods-ci-pr-test/1741/robot/report/files/test_report.html

File: ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/431__data-science-pipelines-api.robot (hunk 20 above)
Comment (jgarciao) [reply ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/431__data-science-pipelines-api.robot:45]: @diegolovison if you added `Remove Pipeline Project` in the keyword teardown to ensure that the project was deleted even if there was a problem deleting the runs o pipeline, that shouldn't be needed because all the keywords of Clear Data Science Pipelines Resources will be run even if one of them fails. The reason is because continue on failure is automatically enabled for Teardown keywords. More info at https://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#continuing-on-failure

File: ods_ci/tests/Resources/Page/Operators/OpenShiftPipelines.resource (hunk 18 above)
Comment (jgarciao) [reply ods_ci/tests/Resources/Page/Operators/OpenShiftPipelines.resource]: Diego moved Tarun's code here so it can be used in different test suites

**Review by jgarciao — APPROVED** [review]
LGTM. Missing suggested fixes are minor and can be added in future PRs

**Review by bdattoma — APPROVED** [review]
